File: app/src/main/java/com/example/sasham/testproject/dependencies/MessagesModule.java
package com.example.sasham.testproject.dependencies;

import com.example.sasham.testproject.messages.MessagesFragment;

import dagger.Module;
import dagger.android.ContributesAndroidInjector;

@Module
public interface MessagesModule {
    @MessagesListingFragmentScope
    @ContributesAndroidInjector(modules = MessagesListingFragmentModule.class)
    MessagesFragment messagesFragmentInjector();
}

File: app/src/main/java/com/example/sasham/testproject/dependencies/MessagesModule.kt
package com.example.sasham.testproject.dependencies

import com.example.sasham.testproject.messages.MessagesFragment

import dagger.Module
import dagger.android.ContributesAndroidInjector

@Module
interface MessagesModule {

    @PerFragment
    @ContributesAndroidInjector(modules = arrayOf(MessagesFragmentModule::class))
    fun messagesFragmentInjector(): MessagesFragment
}
